evennia/contrib/events/README.md

```diff
@@ -0,0 +1,646 @@
+
+## A WARNING REGARDING SECURITY
+
+Evennia's event system will run arbitrary Python code without much restriction.  Such a system is as powerful as potentially dangerous, and you will have to keep in mind two important questions, and answer them for yourself, before deciding to use this system in your game:
+
+1. Is it worth it?  This event system isn't some magical feature that would remove the need for the MU*'s development, and empower immortals to create fabulous things without any control.  Everything that immortals would be able to do through the event system could be achieved by modifying the source code.  Immortals might be familiar with Evennia's design, and could contribute by sending pull requests to your code, for instance.  The event system could admittedly earn you time and have immortals adding in special features without the need for complex code.  You have to consider, however, if it's worth adding this system in your game.  A possible risk is that your immortals will try to do everything though this system and your code will not be updated, while there will still be room to enhance it.
+2. Who should use this system?  Having arbitrary Python code running cannot be considered a secure feature.  You will have to be extremely careful in deciding who can use this system.  By default, immortals can create and edit events (these users have access to the `@py` command, which is potentially as dangerous).  Builders will not be able to add or edit events, although you can change this setting, to have builders be able to create events, and set their events to require approval by an administrator.  You can change permissions (see below for more details on how to do it).  You are free to trust or mistrust your builders or other users, just remember that the potential for malign code cannot be restricted.
+
+## Basic structure and vocabulary
+
+- At the basis of the event system are **event types**.  An **event type** defines the context in which we would like to call some arbitrary code.  For instance, one event type is defined on exits and will fire every time a character traverses through this exit.  Event types are described on a [typeclass](https://github.com/evennia/evennia/wiki/Typeclasses) (like [exits](https://github.com/evennia/evennia/wiki/Objects#exits) in our example).  All objects inheriting from this typeclass will have access to this event type.
+- An event type should specify a **trigger**, a simple name describing the moment when the event type will be fired.  The event type that will be fired every time a character traverses through an exit is called "traverse".  Both "event types" and "trigger" can describe the same thing, although the term **trigger** in the rest of the documentation will be used to describe the moment when the event fires.  Users of the system will be more interested in knowing what triggers are available for such and such objects, while developers will be there to create event types.
```

Comment:
This confuses me, reading as a newcomer to these concepts. Where is the name 'traverse' coming from? How do I know this is what it's called? Is that something you have hardcoded in the event system? I feel there is a confusion of terms here, you are using the term "events" both for the real events fired by the engine (like 'traverse') and for the code callbacks you are assigning to individual objects to *handle* those events. The latter are *not* events and this mixup I think is the reason I have not been able to figure out just how your system is meant to work. Here is how I understand the system, along with what I think would be better names for the components. Correct me if I'm way off:

1. Someone traverses an exit. Evennia(and the event system) fires an *event* of type *traverse*. The event sometimes include arguments relevant for that particular event. 
2. You can add *event handlers* to your typeclasses. An event handler will catch events of a given type (like 'traverse').
3. To these event handlers you in turn attach code *callbacks*. These execute when a given events fire making use of eventual arguments from the event. These callbacks can be different for every instance of the typeclass on which the event handler sits and the code for the callback is added from in-game. 

Note the change in nomenclature here. Trigger to Event, Event type to Event Handler and Event to Callback.